掲示板編集／削除機能(Posts.edit&Posts.delete)

tattu501 opened the pull request:
#何か
投稿一覧、投稿詳細から編集と削除ができるようにする。
<img width="1267" alt="スクリーンショット 2020-11-01 12 41 11" src="https://user-images.githubusercontent.com/61368201/97794732-f0e5aa80-1c40-11eb-9014-1c184ffe34a0.png">
<img width="1280" alt="スクリーンショット 2020-11-01 12 41 37" src="https://user-images.githubusercontent.com/61368201/97794735-ffcc5d00-1c40-11eb-8ffb-872bd919d390.png">
<img width="1280" alt="スクリーンショット 2020-11-01 12 41 52" src="https://user-images.githubusercontent.com/61368201/97794736-022eb700-1c41-11eb-8f5d-e38cd1bf0d14.png">

##追加するファイル
・edit.show.erb

msgyu committed:
RUN "rails g controller posts"

msgyu committed:
Add index in PostsController

indexのviewを作成するために、コントローラーを作成した

msgyu committed:
Add resources posts in routes.rb

indexのview作成のために、追加。
CRUD機能すべてを実装するため、resourcesで記述している。

msgyu committed:
Create index.html.erb of posts

msgyu committed:
Add footer

msgyu committed:
部分テンプレートを読み込み

こちらにfooterを記述する

msgyu committed:
footerを追加

msgyu committed:
Create footer.scss

msgyu committed:
部分テンプレートのパスを変更

sharedフォルダに_footer.html.erbを入れた

msgyu committed:
Edit footer

msgyu committed:
change class in footer

msgyu committed:
Add css in footer.scss

msgyu committed:
階層を変更

msgyu committed:
Add css of a in reset.scss

aタグのリセットcssが記述されていないため、追記した。

msgyu committed:
change color of h4  in footer

tattu501 committed:
routes.rbにresources :postsを追加

ヘッダのビューを確認するため、postsのルーティングを追加

msgyu committed:
Add  column of post model

tattu501 committed:
application.scssにheader.scssのインポートを追加

ヘッダの各ボタンにCSSを反映するため

tattu501 committed:
header.scssにヘッダのSCSSを実装

ヘッダの各ボタンに関して、配置や色を指定するため。

tattu501 committed:
posts.controller.rbにindexアクションを追加

ヘッダの実装状況を確認するため

tattu501 committed:
application.html.erbにheaderの部分テンプレート呼び出しを実装

ヘッダのビューを呼び出すため、posts/header.html.erbをrenderする処理を追加

tattu501 committed:
views/postsにindex.html.erbを追加

ヘッダのビューを確認するため

tattu501 committed:
RUN "rails g controller posts"

ヘッダのビューを確認するためにpostsコントローラを作成した際に出来たファイル

tattu501 committed:
_header.html.erbを作成

ヘッダ用のhtmlを作成するため

ozeryu7 committed:
users_controllerを追加

ozeryu7 committed:
user_controllerを追加した時にできたファイル4つを追加

msgyu committed:
Add Post model

msgyu committed:
Merge branch 'create__model__post' into posts.index

msgyu committed:
define @posts

msgyu committed:
Move index.html.erb

msgyu committed:
Add root in routes.rb

msgyu committed:
Add each in index.html.erb

Display the post repeatedly

msgyu committed:
Add a relation of post and user

msgyu committed:
Merge branch 'create__model__post' into posts.index

msgyu committed:
変数でPostを表示 in index.html.erb

msgyu committed:
投稿順を日付順でDESC

msgyu committed:
Add posts.scss

msgyu committed:
imposrt posts.scss

msgyu committed:
Add includes(:user) to @posts

N+1問題を発生させないために追加

msgyu committed:
投稿者の場合、編集・削除の選択肢を表示

# 何か
投稿者本人の場合、編集・削除ボタンを表示させる。
そんために、post.user_idとcurrent_user.id(ログインuserのid)が一致しているかif文で判定している。

msgyu committed:
Add before_action :authenticate_user! in posts_controller

msgyu committed:
rootの位置を変更

rootを位置に配置した。これでより確認しやすくなった。

msgyu committed:
Merge branch 'header' into posts.index

msgyu committed:
Add link_to

msgyu committed:
Change a to li

msgyu committed:
User認証機能が未完成なため、foreign_keyをコメントアウト

Userの新規登録機能とログイン機能実装後に、コメントアウトを解除する。
migrateしている場合は、rake db:resetを実施してコメントアウト を反映させてから開発します。

tattu501 committed:
headerの修正

headerの余計なコメント（ルートパスへのリンクを設置）を削除、ならびにIF文でログインしているかで表示内容を分岐するように修正しました。

tattu501 committed:
postモデルからuser_idを削除

postモデルのマイグレーションファイルからforeign_keyをコメントアウトされたのを反映するため、rake db:rollbackのあとrake db:migrateを実施

tattu501 committed:
posts_controller.rbにnewアクションを追加

postの登録機能を実装するため、newアクションを追加した

tattu501 committed:
posts/new.html.erbを作成

postの新規投稿ページを実装するため、ビューファイルを作成
※中身はありません。空ファイルを作成

tattu501 committed:
posts/new.html.erbに投稿フォームを実装

postの投稿機能を追加するため、ビューファイルに投稿フォームを実装した

tattu501 committed:
posts.controller.rbからbefore_actionのauthenticate_user!をコメントアウト

新規投稿ページの確認のため、before_action :authenticate_user!, except: [:index, :show]をコメントアウト

tattu501 committed:
posts/shared/_error_messages.html.erbを作成

エラーメッセージ用のパーシャルファイルを作成
※中身はありません

tattu501 committed:
posts/new.html.erbを作成

エラーメッセージ用のパーシャルを除き、新規投稿ページのビューが完成した。

tattu501 committed:
posts/shared/_error_messages.html.erbの中身を作成

新規投稿ページのエラーメッセージを作成

tattu501 committed:
gem 'rails-i18n'をインストール

バリデーションのメッセージを日本語化するため、gem 'rails-i18n'を
Gemfile.lockに追記してbundle installを実施

tattu501 committed:
config/application.rbにエラーメッセージ日本語化用の定義を追加

新規投稿のバリデーションエラー時、メッセージが日本語化されるようにするため、config/application.rbに定義を追加

tattu501 committed:
post.rbに投稿内容のバリデーションを追加

投稿内容がブランク、もしくは140字を超える場合にエラーとするよう、バリデーションを追加した。
なお、belongs_to :userがあると、ユーザ登録機能が未完成のため"userを入れてください。"と出る。バリデーションを正確に検証するため、belongs_to :userをコメントアウトした。

tattu501 committed:
posts.controller.rbにcreateアクションの中身を実装

新規投稿の機能を作成するため、新規投稿用のcreateアクションの内容を実装した。
※前回コミット時にはcreateアクションのガワだけ作成

tattu501 committed:
/posts/shared/_error_messages.html.erbを修正

不要な箇所を削除し、バリデーションエラーのメッセージ実装を完了しました。

tattu501 committed:
config/locales/ja.ymlを追加

バリデーションエラーを日本語するため、ja.ymlを定義した。

ozeryu7 committed:
application.htmlのヘッダーにif user_signed_in?の条件分岐を追加

ヘッダーにサインインしているかで条件分岐を設定

ozeryu7 committed:
application.controllerにストロングパラメーター追加

ストロングパラメーターを追加

ozeryu7 committed:
userモデルにバリデーションを追加

名前、メール、パスワードのバリデーションを追加

ozeryu7 committed:
postテーブル追加時schemaに自動追記

ozeryu7 committed:
config.routesにresourcesを追加

routesにroot to:indexとresources:postsを追加

msgyu committed:
RUN "rails g controller posts"

画面表示するために、土台となるposts_controllerを作成

msgyu committed:
Merge branch 'header' into origin/user.devise

msgyu committed:
Add header

msgyu committed:
partsフォルダに_header.html.erbを移動

_header.html.erbは共通で利用するため、userのログイン時に表示させる必要がある。そのため、共通で呼び出しできる場所に移行した

msgyu committed:
user名をheaderに表示

tattu501 committed:
posts/show.html.erbを作成

投稿詳細ページのビューファイルを作成
※中身はまだありません

tattu501 committed:
posts/show.html.erbにテンプレートを記述

投稿詳細ページのビューにテンプレート内容を記述

tattu501 committed:
posts/show.html.erbの投稿一覧ページへのリンクを作成

投稿一覧ページに戻るボタン「一覧へ戻る」にリンクを付与

msgyu committed:
Merge branch 'origin/user.devise' into posts.create

msgyu committed:
_header.html.erbをsharedフォルダに移行

footerが格納されているsharedにheader.html.erbを入れ、共通フォルダに入れた

msgyu committed:
RUN"rails db:migrate:reset"

userモデルのforeign_keyを追加した

msgyu committed:
before_action :authenticate_user!のコメントアウトを解除

Userモデルを追加したので、 before_action :authenticate_user!が利用できるようになった。そのためにコメントアウト を解除した

msgyu committed:
コメント数を数字で表示

tattu501 committed:
Merge branch 'origin/user.devise' into posts.create

msgyu committed:
コメント・いいね数を数字で表示

仮置きとして１を記述した

tattu501 committed:
post.rbのbelongs_to :userを再実装

belongs_to :userユーザのコメントアウトを外し、userモデルとpostモデルを紐付ける

tattu501 committed:
postモデルに外部キーuser_idを再実装

rake db:resetを行い、外部キーuser_idを再度実装した

tattu501 committed:
posts.controller.rbにbefore_actionを再実装

コメントアウトしていたbefore_action :authenticate_user!, except: [:index, :show]を再度実装した

tattu501 committed:
_header.html.erbの「投稿する」ボタンにリンクを付与

「投稿する」ボタンに新規投稿ページに遷移するリンクを付与した。

msgyu committed:
Add gem 'pry-rails'

msgyu committed:
@postsのincludes(:user)を先に記述

先に呼び出すことで処理速度を向上させるため

msgyu committed:
= を追加して投稿情報を表示

msgyu committed:
=を追加して投稿情報を表示

msgyu committed:
postをlink_toの中に格納

これでpostのdivをクリックするだけでposts.showに移行する。

msgyu committed:
Merge branch 'posts.index' into posts.create

msgyu committed:
show.html.erbをpostsフォルダに移行

msgyu committed:
@post を定義 in posts.show

msgyu committed:
show.html.erbにおいて@postを挿入し、カラム情報を表示

msgyu committed:
import _comments.scss

msgyu committed:
一覧へ戻るボタンにpathを追加

下記のようにlink_toの中にボタンのhtmlを入れることで、一覧へ戻るをクリックするだけで戻ることができる。

 <%= link_to root_path do%>
    ボタンのhtml
<% end %>

msgyu committed:
= を追加して投稿日時を表示

msgyu committed:
time_zonをtokyoに設定

日付日時を日本時間で表示させるために、追加

msgyu committed:
日本日時で表示させるため、日時フォーマットを追加

msgyu committed:
日時フォーマートを変更

msgyu committed:
投稿者でなければ編集・削除が表示されない

<% if @post.user_id == current_user.id%>
postの外部キーuser_idと現在ログイン中のuser.idが一致している場合に表示される。

tattu501 committed:
postモデルの外部キーuser_idを再実装

user認証が完成したので、postモデルに元々あった外部キーuser_idを再度実装した

tattu501 committed:
post.controller.rbのcreateアクションをuserモデルと紐付けた

user認証が完成したので、Post.newからcurrent_user.posts.buildに変更し、postの外部キーuser_idにユーザのIDが入るようにした。

tattu501 committed:
posts/index.html.erbに投稿内容を表示できるようにした

新規投稿機能が完成したので、投稿した内容が一覧ページに表示されるようにした。

tattu501 committed:
pullした内容を反映

pullした最新の内容を反映

tattu501 committed:
pullした内容を反映

pullした最新の内容を反映

tattu501 committed:
application.html.erbの不要な箇所を削除

applicaiton.html.erbに不要な文字列が残っていたので削除した

tattu501 committed:
Merge branch 'posts.create' into posts.show

tattu501 committed:
posts/index.html.erbにログインしているかのIF文を追加

未ログイン時に<% if post.user_id == current_user.id %>でidがないとしてエラーになる。エラー回避のため、ログイン済みの場合のみ<% if post.user_id == current_user.id %>の判定が行われるように考慮を追加

tattu501 committed:
posts.controller.rbにshowアクションを追加

投稿詳細ページに内容が表示されるよう、データを読み込む処理を追加

tattu501 committed:
posts/show.html.erbに投稿内容を表示

投稿内容をビューに表示できるよう、各項目を埋め込みrubyで出力する　※コメントは未実装のため固定値で表示

tattu501 committed:
posts/show.html.erbに自ユーザのみ編集、削除ボタンを表示

投稿詳細を閲覧するユーザと投稿ユーザが一致する場合のみ、「編集」「削除」ボタンを表示する

tattu501 committed:
_header.html.erbにflashメッセージの出力を追加

before_action用のflashメッセージを出力出来るようにコードを追加

tattu501 committed:
posts/edit.html.erbのファイルを追加

掲示板を編集・削除できるのを投稿者のみに限定した。その確認のため、editビューの空ファイルを追加した。

tattu501 committed:
posts.controler.rbにbefore_actionのauthenticate_admin!を追加

投稿者以外が編集ページにアクセスしようとすると、投稿一覧ページへ遷移するよう、authenticate_admin!メソッドをbefore_actionとして実装した。

tattu501 committed:
Merge branch 'posts.show' of https://github.com/ozeryu7/tenkue-board-r006 into posts.show

tattu501 committed:
posts/show.html.erbを修正

未ログイン時に投稿一覧ページに行くと、current_user.idが存在しないので、投稿者とログインユーザが一致するかの判定でエラーが発生していた。そのため、投稿者とログインユーザの判定にはログイン時のみ入るように修正した。

tattu501 committed:
config/locales/devise.ja.ymlを追加

deviseのメッセージを日本語化するため

msgyu committed:
ログアウト時のエラーを防ぐため、if user_signed_in?を追加

ログアウト時はcurrent_userの情報を取得できないため、エラーが発生する。そのため、if user_signed_in?でログインしているか判定している

msgyu committed:
Merge branch 'posts.index' into posts.create

msgyu committed:
Revert "pullした内容を反映"

This reverts commit 30c51af9a7aa13072bf1c9f0034ec3ff95876ee5.

msgyu committed:
チームメンバーのpullを修正

msgyu committed:
チームメンバーのpullを修正

msgyu committed:
チームメンバーのpullを修正

msgyu committed:
Revert "チームメンバーのpullを修正"

This reverts commit f77ab5a174f5306701ad8c203bdbeabcad327cc7.

msgyu committed:
Revert "チームメンバーのpullを修正"

This reverts commit 67e59edafbd8d0a6db44411b83754e7fc3883674.

msgyu committed:
Revert "チームメンバーのpullを修正"

This reverts commit e22044c2977da92e6b4f2ac536397a49e93f8335.

msgyu committed:
Revert "Revert "pullした内容を反映""

This reverts commit 63fdcfeb04cdec41e747d3dda9df623f5a9e16f0.

tattu501 committed:
Merge branch 'posts.create' into posts.edit/delete

tattu501 committed:
posts/index.html.erbに投稿の編集、削除リンクを追加

投稿の編集ページに遷移するリンク、ならびに削除するリンクを追加

tattu501 committed:
posts/show.html.erbに編集、削除リンクを追加

投稿詳細ページに編集リンク、削除リンクを追加

tattu501 committed:
posts/show.html.erbで投稿の編集・削除ができるのは投稿者のみとする

掲示板を編集・削除できるのは投稿者のみとする

tattu501 committed:
templates/edit.html.erbを削除

タイトルのテンプレートファイルは不要なので削除

tattu501 committed:
posts/show.html.erbを修正

間違えてコメントの編集・削除ボタンを書き換えてしまったので、変更を元に戻し、改めて投稿の編集・削除ボタンを実装した。

tattu501 committed:
posts/edit.html.erbの内容を実装

投稿編集ページのビューに内容を実装

tattu501 committed:
posts.controller.rbにedit/update/destroyの3アクションを追加

投稿編集・削除の機能を実装するため、edit/update/destroyの合計3アクションを追加

msgyu commented on app/assets/javascripts/posts.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
不要なファイルの削除

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on app/assets/javascripts/users.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
+# You can use CoffeeScript in this file: http://coffeescript.org/
```
不要なファイルの削除

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on app/assets/stylesheets/posts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the posts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
不要なファイルの削除

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on app/assets/stylesheets/users.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
不要なファイルの削除

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on db/migrate/20201030162743_add_user_id_to_posts.rb:
```diff
@@ -0,0 +1,4 @@
+class AddUserIdToPosts < ActiveRecord::Migration[5.2]
+  def change
+  end
+end
```
不要なマイグレーションファイル

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on db/schema.rb:
```diff
@@ -10,11 +10,19 @@
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_08_08_145604) do
+ActiveRecord::Schema.define(version: 2020_10_30_162743) do
```
不要なマイグレーションファイルを削除して、rake db:reset

tattu501 replied:
申し訳ございません。
マイグレーションファイル削除後にrake db:resetしましたが、日時が変わりませんでした。

msgyu replied:
無事に動作し致命傷なるものではないので、このまま進めましょう

msgyu commented on app/views/posts/show.html.erb:
```diff
@@ -4,45 +4,48 @@
-              <span>いいね数</span>
+              <span>1</span>
             </li>
+            <% if user_signed_in? && @post.user_id == current_user.id %>
+              <div class="post_detail__bottom_box__admin_actions">
+                <%= link_to "編集", edit_post_path(@post.id) %>
+                <%= link_to "削除", @post, method: :delete, data: { confirm: '本当に削除して良いですか?', cancel: 'やめる', commit: '削除する'}, title: '削除確認' %>
+              </div>
+            <% end %>
           </ul>
```
view崩れ
/ulの外に移動する必要があります。
元々はulのあとに　div class="post_detail__bottom_box__admin_actions"が存在していたが順序が逆になっている。
rails s でページを確認しながら修正してください
文字の色も青色ではないので、そこも確認しておきましょう

tattu501 replied:
ulのあとに　div class="post_detail__bottom_box__admin_actions"を移動しました。
また文字を青色にしました。

tattu501 committed:
不要なファイルを削除

不要なファイルを削除しました。
・app/assets/javascripts/posts.coffee
・app/assets/javascripts/users.coffee
・app/assets/stylesheets/posts.scss
・app/assets/stylesheets/users.scss
・db/migrate/20201030162743_add_user_id_to_posts.rb

tattu501 committed:
Revert "不要なファイルを削除"

This reverts commit c85663fc158c1e165de6259ec8d06d22ef9d57ba.

tattu501 committed:
posts/show.html.erbを修正　

ulのあとにdiv class="post_detail__bottom_box__admin_actions"に移動した。

tattu501 committed:
不要なファイルを削除

不要なファイルをしました。
・app/assets/javascripts/posts.coffee
・app/assets/javascripts/users.coffee
・app/assets/stylesheets/posts.scss
・app/assets/stylesheets/users.scss
・db/migrate/20201030162743_add_user_id_to_posts.rb

tattu501 committed:
posts.scssを修正

投稿詳細ページの編集、削除ボタンが青色になるよう、SCSSを修正

msgyu commented:
LGTM

masato-sa93 commented on app/controllers/application_controller.rb:
```diff
@@ -1,2 +1,11 @@
+  protected
+
+  def configure_permitted_parameters
+    added_attrs = [ :name, :email, :password ]
```
ここのkey指定はnameだけで大丈夫なはずですー！
（devise側でemailとpasswordは標準で準備してくれているので）

※列挙しても大丈夫と言えば大丈夫で好みの問題なので、修正しなくてもOKです

tattu501 replied:
key指定をnameのみに変更しました。

masato-sa93 commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :authenticate_admin!, except: [:index, :show, :new, :create]
```
exceptで4つアクションを指定するなら、onlyで3つ指定してあげた方が可読性高いと思いますー！

exceptは「〜を除く」で逆説的な意味合いなので、その文脈に合った使い方をすると読みやすいコードになります！

tattu501 replied:
exceptからonlyに修正しました。

nousi replied:
before_action :authenticate_admin!, except: [:index, :show, :new, :create]は不要なので、削除が必要です。
before_action :authenticate_user!, except: [:index, :show]をそのまま利用すれば問題ないです

masato-sa93 commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,58 @@
+    end
+
+    def authenticate_admin!
+      @post = Post.includes(:user).find(params[:id])
```
単一のpostインスタンスに、includesを記述してる意図って何かありますか？？

tattu501 replied:
includes(:user)は間違えて記述しておりました。
不要なので削除しました。

masato-sa93 commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
```
usersコントローラーは何かに使ってますか？

tattu501 replied:
使っている先はないです。不要なのでファイルを削除しました。

masato-sa93 commented on app/views/posts/edit.html.erb:
```diff
@@ -0,0 +1,22 @@
+      <h3>
+        <span>投稿編集</span>
+      </h3>
+      <%= form_for(@post) do |f| %>
```
Rails5.2以上の場合は`form_with`の方がベストプラクティスですー！

※`form_for`でも問題はないので、修正はしなくても大丈夫です。

tattu501 replied:
ご教示頂きありがとうございます。承知しました。

masato-sa93 commented on app/views/shared/_header.html.erb:
```diff
@@ -0,0 +1,32 @@
+<header class="header">
```
headerのビューファイルが2つできているので、不要な方を削除しましょう！

tattu501 replied:
app/views/posts/_header.html.erbが不要なので削除しました。

masato-sa93 reviewed (changes requested):
レビュー完了しました。
修正後、再度依頼をお願いしますー！

msgyu commented on app/views/posts/_header.html.erb:
```diff
@@ -0,0 +1,28 @@
+<header class="header">
+  <div class="header__bar">
+    <h1>
+      <%= link_to "転クエ掲示板", root_path %>
+    </h1>
+    <%# ログイン状態に応じて表記を変更すること %>
+    <ul class="header__bar__user_nav">
+      <li class="user_name">
+        <span>ゲストユーザ さん</span>
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "ログアウト", "#" %>
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "投稿する", "#" %>
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "投稿する", "#" %>    
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "ログイン", "#" %> 
+      </li>
+      <li class="header__bar__user_nav__btn">
+        <%= link_to "新規登録", "#" %> 
+      </li>
+    </ul>
+  </div>
+</header>
```
partsのheader.html.erbを削除

tattu501 replied:
上記ファイルを削除しました。

msgyu commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,11 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def new
+    @user = User.new
+  end
+
+  
+end
```
users_controller.rbを削除

tattu501 replied:
上記ファイルを削除しました。

tattu501 committed:
application_controller.rbを修正

added_attrsのkey指定をnameのみに変更

tattu501 committed:
posts.controller.rbを修正

before_action :authenticate_admin!のアクション指定をexceptからonlyに変更

tattu501 committed:
posts.controller.rbを修正

authenticate_admin!でpostの内容を取得するとき、includes(:user)が不要だったので削除しました。

tattu501 committed:
users.controller.rbを削除

不要なので削除

tattu501 committed:
posts/_header.html.erbを削除

不要なので削除

nousi commented on app/controllers/posts_controller.rb:
```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :authenticate_admin!, only: [:edit, :update, :destroy]
```
before_action :authenticate_admin!, only: [:edit, :update, :destroy]を削除しましょう

before_action :authenticate_admin!, only: [:edit, :update, :destroy]
＞今回はuserモデルなので、before_action :authenticate_user!, except: [:index, :show]で統一しましょう。

before_action :authenticate_admin!の場合は、adminモデルである必要があります。
また、index,show以外はログイン認証が必要なので、

before_action :authenticate_user!, except: [:index, :show]だけで大丈夫です。

tattu501 committed:
posts.controller.rbを修正

before_action :authenticate_admin!, only: [:edit, :update, :destroy]を削除

tattu501 committed:
posts.controller.rbの修正を取り消し

before_action :authenticate_admin!, only: [:edit, :update, :destroy]を直前に削除したが、該当のコメントが削除されたので修正

masato-sa93 reviewed (approved):
LGTMですー！
さすがです！修正お疲れ様でした
